integration-tests/test-extension/extension/deployment/src/test/java/io/quarkus/config/SecretKeysConfigInterceptorFactory.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.config;
+
+import java.util.Set;
+
+import io.smallrye.config.ConfigSourceInterceptor;
+import io.smallrye.config.ConfigSourceInterceptorContext;
+import io.smallrye.config.ConfigSourceInterceptorFactory;
+import io.smallrye.config.SecretKeysConfigSourceInterceptor;
+
+public class SecretKeysConfigInterceptorFactory implements ConfigSourceInterceptorFactory {
+    @Override
+    public ConfigSourceInterceptor getInterceptor(final ConfigSourceInterceptorContext configSourceInterceptorContext) {
+        return new SecretKeysConfigSourceInterceptor(Set.of("secrets.my.secret"));
```

Comment:
The future idea is to introduce some types that represent the secrets, so you use that instead in ConfigRoot / Mappings. In that case, you shouldn't need to register keys individually.

Reply:
@radcortez For user-supplied secrets not corresponding to a config key known by Quarkus, it could be interesting to have a build-time property to list these secrets. Something like `quarkus.config.secrets=secrets.my.secret`.

Reply:
Well if you use a `@ConfigMapping` we can either provide a type that represents the secret, or even provide some extra annotation to mark a specific property as being a secret, so you don't need to manually list the keys one by one.